### PR TITLE
Fixes next error and starts column logic

### PR DESCRIPTION
**Pull request by zacjones93**
Because I'm using react-beautiful-dnd with Next I ran into an error where dnd couldn't find the drag handle for items. This has to do with SSR issues that I don't full understand.

This fixed my issue:
https://github.com/atlassian/react-beautiful-dnd/issues/2092

Essentially, I need to ensure that the window is ready before rendering anything with react-beautiful-dnd.

I also started to implement the drag and drop logic that I will need for multi-column support

columns

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -48,35 +47,53 @@ const BugList = ({bugs}: {bugs: any[]}) => {
 
 
 const Home: NextPage = (props) => {
+  const [winReady, setwinReady] = useState(false);
+  useEffect(() => {
+      setwinReady(true);
+  }, []);
```
Comment (zacjones93) [inline pages/index.tsx:53]: Window will be ready after useEffect runs so this sets a state variable to true

File: pages/index.tsx [hunk 2]
```diff
@@ -86,14 +103,14 @@ const Home: NextPage = (props) => {
       </Head>
 
       <main className="flex flex-col items-center justify-center flex-1 w-full px-20 text-center bg-slate-500">
-        <DragDropContext onDragEnd={onDragEnd}>
+        {winReady ? <DragDropContext onDragEnd={onDragEnd}>
```
Comment (zacjones93) [inline pages/index.tsx:106]: only renders DragDropContext when winReady is set

File: pages/index.tsx [hunk 3]
```diff
@@ -48,35 +47,53 @@ const BugList = ({bugs}: {bugs: any[]}) => {
...
     if (!result.destination) {
       return;
     }
 
     if (result.destination.index === result.source.index) {
       return;
     }
-
+    
     if (result.destination.droppableId === result.source.droppableId) {
```
Comment (zacjones93) [inline pages/index.tsx:67]: Checks to see if destination and source are the same and if so, apply the same single column logic that we've been using.

File: pages/index.tsx [hunk 4]
```diff
@@ -48,35 +47,53 @@ const BugList = ({bugs}: {bugs: any[]}) => {
...
+      },
+      "newCol: ", {newCol}
+      )
+      
+      setState({ 
+        ...state,
+        [destinationColumn.id]: {
+          ...newCol
+        }
+    });
```
Comment (zacjones93) [inline pages/index.tsx:93]: re-creates a column and sets state with previous state spread and that column set with the new column replacing the previous column.

effectively this reorders the bug list

File: pages/index.tsx (hunk 2 above)
Comment (valendion) [reply pages/index.tsx:106]: it works for me, but what is the condition "else"?
is it null?

**Pull request merged**